src/Native/Runtime/windows/CoffNativeCodeManager.cpp

```diff
@@ -380,9 +380,82 @@ void CoffNativeCodeManager::EnumGcRefs(MethodInfo *    pMethodInfo,
 
 UIntNative CoffNativeCodeManager::GetConservativeUpperBoundForOutgoingArgs(MethodInfo * pMethodInfo, REGDISPLAY * pRegisterSet)
 {
-    // @TODO: CORERT: GetConservativeUpperBoundForOutgoingArgs
+#if defined(_TARGET_AMD64_)
+
+    // Return value
+    UIntNative upperBound;
+    CoffNativeMethodInfo* pNativeMethodInfo = (CoffNativeMethodInfo *) pMethodInfo;
+    
+    size_t unwindDataBlobSize;
+    PTR_VOID pUnwindDataBlob = GetUnwindDataBlob(m_moduleBase, pNativeMethodInfo->runtimeFunction, &unwindDataBlobSize);
+    PTR_UInt8 p = dac_cast<PTR_UInt8>(pUnwindDataBlob) + unwindDataBlobSize;
+    uint8_t unwindBlockFlags = *p++;
+
+    if ((unwindBlockFlags & UBF_FUNC_HAS_ASSOCIATED_DATA) != 0)
+        p += sizeof(int32_t);
+
+    if ((unwindBlockFlags & UBF_FUNC_REVERSE_PINVOKE) != 0)
+    {
+        TADDR basePointer =  dac_cast<TADDR>(pRegisterSet->GetFP());        
+        
+        // Get the method's GC info
+        GcInfoDecoder decoder(GCInfoToken(p), DECODE_REVERSE_PINVOKE_VAR);
+        UINT32 stackBasedRegister = decoder.GetStackBaseRegister();
+        
+        if (stackBasedRegister == NO_STACK_BASE_REGISTER)
+        {
+            basePointer = dac_cast<TADDR>(pRegisterSet->GetSP());
+        }
+        else
+        {
+            basePointer = dac_cast<TADDR>(pRegisterSet->GetFP());
+        }
+        // Reverse PInvoke case.  The embedded reverse PInvoke frame is guaranteed to reside above
+        // all outgoing arguments.
+        INT32 slot = decoder.GetReversePInvokeFrameStackSlot();
+        upperBound =  (UIntNative) dac_cast<TADDR>(basePointer + slot);
+    }
+
+    else
+    {
+        bool rbp = GetFramePointer(pMethodInfo, pRegisterSet) == NULL;
+        if (!rbp) 
+        {
+            // Unwind the current method context to get the caller's stack pointer
+            // and obtain the upper bound of the callee is the value just below the caller's return address on the stack
+            SIZE_T  EstablisherFrame;
+            PVOID   HandlerData;
+            CONTEXT context;
+            context.Rsp = pRegisterSet->GetSP();
+            context.Rbp = pRegisterSet->GetFP();
+            context.Rip = pRegisterSet->GetIP();
+    
+            RtlVirtualUnwind(NULL,
+                            dac_cast<TADDR>(m_moduleBase),
+                            pRegisterSet->IP,
+                            (PRUNTIME_FUNCTION)pNativeMethodInfo->runtimeFunction,
+                            &context,
+                            &HandlerData,
+                            &EstablisherFrame,
+                            NULL);
+
+            upperBound = dac_cast<TADDR>(context.Rsp - 2 * sizeof (PVOID));
```

Comment:
It is still not clear to me where the `2 * sizeof (PVOID)` came from.

https://github.com/dotnet/corert/pull/5585#discussion_r178227509